src/Components/Web/src/Forms/EditForm.cs

```diff
@@ -91,9 +106,9 @@ protected override void OnParametersSet()
 
             // Update _fixedEditContext if we don't have one yet, or if they are supplying a
             // potentially new EditContext, or if they are supplying a different Model
-            if (_fixedEditContext == null || EditContext != null || Model != _fixedEditContext.Model)
+            if (_fixedEditContext == null || (!_hasSetEditContextExplicitly && Model != _fixedEditContext.Model))
```

Comment:
Nit: I think this could be simplified to:

```suggestion
            if (!_hasSetEditContextExplicitly && Model != _fixedEditContext?.Model)
```

...  because if `_hasSetEditContextExplicitly` then we know `_fixedEditContext != null`.

Reply:
Also the comment above it is slightly out of date now

Reply:
Strictly speaking the check could be expressed as:

```cs
if (Model != null && Model != _fixedEditContext?.Model)
```

(because we know that if `Model != null`, then `_hasSetEditContextExplicitly` is false otherwise we would already have thrown above)

Arguably this is a more natural phrasing because it's just stating that if you give us any new model, we want to use it.